tests/platform_tests/counterpoll/test_counterpoll_watermark.py

```diff
@@ -0,0 +1,233 @@
+            pytest_assert(map_output, msg)
+            for line in map_output:
+                try:
+                    map.remove(line)
+                except ValueError:
+                    failed = "MAP {} was not found in {} MAPS list".format(line, map_prefix)
+            pytest_assert("" == failed, failed)
+            pytest_assert(len(map) == 0, "{} maps mismatch, one or more queue was not found in redis COUNTERS_DB"
+                          .format(map_prefix))
+
+    failed_list = []
+    with allure.step("Verifying {} STATS in FLEX_COUNTER_DB on {}...".format(tested_counterpoll, duthost.hostname)):
+        stats_output = redis_get_keys(duthost, 'FLEX_COUNTER_DB', '*{}*'.format(map_prefix))
+        counted = 0
+        # build expected counterpoll stats vs unexpected
+        expected_types = []
+        unexpected_types = []
+        for counterpoll, v in RELEVANT_MAPS.items():
+            types_to_check = v[CounterpollConstants.TYPE]
+            if counterpoll in tested_counterpoll:
+                for type in types_to_check:
+                    expected_types.append(FLEX_COUNTER_PREFIX + type)
+            else:
+                for type in types_to_check:
+                    unexpected_types.append(FLEX_COUNTER_PREFIX + type)
+        logging.info("expected types for for counterpoll {}:\n{}".format(tested_counterpoll, expected_types))
+        logging.info("unexpected types for for counterpoll {}:\n{}".format(tested_counterpoll, unexpected_types))
+        for line in stats_output:
+            for expected in expected_types:
+                if expected in line:
+                    counted += 1
+            for unexpected in unexpected_types:
+                if unexpected in line:
+                    failed_list.append("found for {} unexpected stat counter in FLEX_COUNTER_DB: {}"
+                                       .format(tested_counterpoll, line))
+        logging.info("counted {} {} STATs type in FLEX_COUNTER_DB on {}..."
+                     .format(counted, tested_counterpoll, duthost.hostname))
+        failed_list
+        pytest_assert(counted > 0, "counted {} for {}".format(counted, tested_counterpoll))
+
+    # for watermark only, also count stats with actual values in COUNTERS_DB
+    if CounterpollConstants.WATERMARK in tested_counterpoll:
+        with allure.step("counting {} STATS in FLEX_COUNTER_DB on {}...".format(tested_counterpoll, duthost.hostname)):
+            count_watermark_stats_in_counters_db(duthost)
+
+    pytest_assert([] == failed_list, failed_list)
+
+    # no need for reload or reboot when enabling all queue/watermark/pg-drop counterpolls
+    with allure.step("enable all {} counterpolls on {} and check results..."
+                     .format(RELEVANT_COUNTERPOLLS, duthost.hostname)):
+        ConterpollHelper.enable_counterpoll(duthost, RELEVANT_COUNTERPOLLS)
+    verify_counterpoll_status(duthost, RELEVANT_COUNTERPOLLS, ENABLE)
```

Comment:
Same as above. Should `verify_counterpoll_status(duthost, RELEVANT_COUNTERPOLLS, ENABLE)` in the with block?

Reply:
same here, verify_counterpoll_status isn't part of the enablement but can be aligned under this with block and change the comment printed accordingly

Reply:
Done